### PR TITLE
Correct the set item logic of `CodeCellModel.onModelDBOutputsChange`

### LINKED ISSUE #12131: Strange behavior of displaying cell with multiple outputs in collaborative mode.
## Description

In collaborative mode, If a client executes a cell that contains more than one display output and other clients trust their notebook, the output of the original client is modified. The last displayed item will be duplicated in the first client.

https://user-images.githubusercontent.com/4451292/156052858-7a4a9e3e-d2a2-49f2-8e35-dde15205de5d.mp4

## Reproduce

- Start jupyter lab in collaborative mode
- Create a cell with two display outputs:
```python
import ipywidgets as ipw
display(ipw.FloatSlider())
print('bar')
```
- Execute the cell in one client, trust the notebook on another.

## Expected behavior

The output of the first client should not be modified. 

## Context

- JupyterLab version: 3.2.9

### PR DESCRIPTION
**Pull request by trungleduc**
## References
close  #12131 

## Code changes

- Correct the index value of items in `CodeCellModel.onModelDBOutputsChange`
-  Add tests for  `CodeCellModel.onModelDBOutputsChange`

## User-facing changes
N/A

## Backwards-incompatible changes
N/A

### CODE REVIEW

File: packages/cells/test/model.spec.ts [hunk 1]
```diff
@@ -541,6 +542,111 @@ describe('cells/model', () => {
...
+        expect(sharedModel.ymodel.get('outputs').length).toBe(0);
+
+        const newEvent0 = {
+          type: 'add',
+          newValues: [{ toJSON: () => output0 }],
+          oldValues: [],
+          oldIndex: -1,
+          newIndex: 0
+        } as any;
+        model['onModelDBOutputsChange'](null as any, newEvent0);
```
Comment (fcollonval) [inline packages/cells/test/model.spec.ts:590]: Is there a reason for using a key getter and not

```suggestion
        model.onModelDBOutputsChange(null as any, newEvent0);
```

**Review by fcollonval — COMMENTED** [review]
Thanks @trungleduc 

i got one question

File: packages/cells/test/model.spec.ts (hunk 1 above)
Comment (trungleduc) [reply packages/cells/test/model.spec.ts:590]: Hello @fcollonval , it's because `onModelDBOutputsChange` is a protected method.

Comment (fcollonval) [reply packages/cells/test/model.spec.ts:590]: thanks - I didn't look at the code :wink:

**Review by fcollonval — APPROVED** [review]

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.3.x

**Pull request merged**